### ISSUE TITLE
[Question] Contents of train.json for *.tgz in https://katagoarchive.org/kata1/trainingdata/index.html

### ISSUE THREAD

**Issue by y-ich** [lightvector/KataGo#1001]
Hi.

I want to train a weight using *.tgz in https://katagoarchive.org/kata1/trainingdata/index.html.
What is the content of train.json for each tgz files?
They do not include it.

Thanks!

**Comment by lightvector** [comment]
For these files, there is no such thing. "train.json" is output by the script that shuffles the data. These files have never been shuffled, they are the raw files uploaded by users for different games, so there is no json. You can shuffle the data yourself, see python/selfplay/shuffle.sh.

**Comment by y-ich** [comment]
lightvector san, thank you for your kind comment!

I shuffled data and succeeded to start training but stopped due to out-of-memory.
I am trying training on my 16GB MacBook Air.
Could you suggest to me how much GPU memory you need for training of b28c512nbt?
I may rent the one I need.

Thanks!

**Comment by lightvector** [comment]
Try reducing batch size a little at a time, generally you should use the close to the largest batch size you have GPU memory for. Also you can try different batch sizes and measure the see the GPU performance, different GPUs can sometimes have sudden sharp cliffs in performance where there are critical batch sizes where crossing that threshold suddenly makes performance better or worse, it's not *always* the case that the largest batch size that fits in memory is best.

Unless you set the batch size extremely small, generally training with different batch sizes should be relatively similar once the neural net has converged a little bit. Very tiny batch size will probably train poorly because batch norm relies on averaging across batches. Extremely huge batch size might be more unstable unless the learning rate is decreased. But mostly batch size doesn't matter too much. Maybe it has small long term effects on quality but I don't know what those are because I haven't done experiments to measure that.

**Comment by y-ich** [comment]
Thank you!

First I tried b28c512nbt but it seems  to need long time for training.
And I tried b5c192nbt but it also does not end in one day on my M2 MacBook Air.

How much time do you typically need for training on your GPU?

Excuse me for asking basic questions.
Thank you!

**Comment by lightvector** [comment]
If you wanted to re-train b28c512nbt from scratch using the existing training data, it would probably take around 8 months of continuous non-stop training on 2 x A100. So yes, there is a lot of data and you need a lot of GPU power to train them.

**Comment by y-ich** [comment]
Sorry for my ambiguous question.

What I want to know is the typical time for just weight training, which does not include selfplays.
Two days seem not enough for b5c192nbt training on my M2 MBA...

About background.
Apple silicon (A17 Pro, M4, A18) now support INT8 acceleration.
But it needs quantization-aware training.
So I want to train a neural network by quantization-aware training using katagotraining.org selfplay data.

Thanks.

**Comment by y-ich** [comment]
Are you spending such huge resources to generate each b28c512nbt weight??
And how much resources about b5c192nbt?

(About background.
Apple silicon (A17 Pro, M4, A18) now support INT8 acceleration.
But it needs quantization-aware training.
So I want to train a neural network by quantization-aware training using katagotraining.org selfplay data.)

Thanks.

**Comment by lightvector** [comment]
> Are you spending such huge resources to generate each b28c512nbt weight??
Of course not, that's the *total* training cost if you restarted the network from scratch on the current self-play data. But each new weight file uploaded is NOT a new network started from scratch, instead it's the same network with a little bit more training each time. See the "s" number in the name kata1-b28c512nbt-s8003120896-d4541551568, and how it rises with each file, that "s" number is the number of training steps for that same network so far. 8003120896.

I've never trained b5c192nbt before, but you can do the math yourself. See how fast it is on your machine in how many samples it trains on in a given time "nsamp", estimate what "s" number of trained samples you want to reach (such as 8003120896 in kata1-b28c512nbt-s8003120896-d4541551568) and multiply.

As I said before, if you want to train a new network from scratch it will take lot of GPU power. Even a smaller network will take a decent amount of power, and a full-size one as I said above will take months even on powerful GPUs.
